main/lsp/LSPTypechecker.cc

```diff
@@ -334,20 +335,15 @@ LSPTypechecker::SlowPathResult LSPTypechecker::runSlowPath(LSPFileUpdates &updat
     auto slowPathOp = std::make_optional<ShowOperation>(*config, ShowOperation::Kind::SlowPathBlocking);
     Timer timeit(logger, "slow_path");
     ENFORCE(updates.typecheckingPath != TypecheckingPath::Fast || config->disableFastPath);
-    ENFORCE(updates.updatedGS.has_value());
-    if (!updates.updatedGS.has_value()) {
-        Exception::raise("runSlowPath called with an update that lacks an updated global state.");
-    }
     logger->debug("Taking slow path");
 
     ENFORCE(this->cancellationUndoState == nullptr);
     if (cancelable) {
+        auto savedGS = std::exchange(this->gs, pipeline::copyForSlowPath(*this->gs, this->config->opts));
```

Comment:
(This call occurs above `pipeline::reserveFiles` which is why in my other suggestion, I think this function should live in the top section in pipeline.cc)